llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -4254,10 +4254,6 @@ SDValue AMDGPUTargetLowering::performMulCombine(SDNode *N,
     return DAG.getNode(ISD::ADD, DL, VT, MulVal, N0);
   }
 
-  // Skip if already mul24.
-  if (N->getOpcode() != ISD::MUL)
```

Comment:
Probably should turn this into an assert? Is there definitely nothing else in performMulCombine that can operate on mul24?